### PR TITLE
Release tracking PR: `v12.0.0`

### PR DESCRIPTION
**Pull request by tcharding**
In preparation for release add a changelog entry and bump the version.

### CODE REVIEW

**Comment by tcharding** [comment]
cc @notmandatory

**Comment by tcharding** [comment]
@apoelstra and/or @sanket1729 can you guys please look closer than usual at the changelog entries because I don't know what goes on here as closely as I do in other crates, specifically are the sub-headings correct and/or useful? I basically included every merged PR since last release.

Thanks

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,30 @@
...
+
+- Update MSRV to Rust 1.56.1 [#639](https://github.com/rust-bitcoin/rust-miniscript/pull/639)
+- Drop the `Property` trait entirely [#652](https://github.com/rust-bitcoin/rust-miniscript/pull/652)
+- Improve compiler logic when deciding between conjunctions and multi/multi_a [#657](https://github.com/rust-bitcoin/rust-miniscript/pull/657)
+- Several locktime improvements [#654](https://github.com/rust-bitcoin/rust-miniscript/pull/654)
+- Derives `Hash` for pub items [#659](https://github.com/rust-bitcoin/rust-miniscript/pull/659)
+- Introduce `Threshold` type [#660](https://github.com/rust-bitcoin/rust-miniscript/pull/660), 
+  [#674](https://github.com/rust-bitcoin/rust-miniscript/pull/674),
+  and [#676](https://github.com/rust-bitcoin/rust-miniscript/pull/676)
+- Upgrade `bech32` dependency to `v0.11.0` [#661](https://github.com/rust-bitcoin/rust-miniscript/pull/661)
```
Comment (storopoli) [inline CHANGELOG.md:11]: Either this version is non code, i.e. 0.11.0 or the MSRV Rust version above should be as code (backticks) for consistency

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,30 @@
...
+- Return `Weight` type for `max_weight_to_satisfy` methods [#664](https://github.com/rust-bitcoin/rust-miniscript/pull/664)
+
+## Performance/compiled time improvements
+
+- Remove recursion in `semantic` module [#612](https://github.com/rust-bitcoin/rust-miniscript/pull/612)
+- Remove generics from Error by making fragment a String [#642](https://github.com/rust-bitcoin/rust-miniscript/pull/642)
```
Comment (storopoli) [inline CHANGELOG.md:17]:
```suggestion
- Remove generics from `Error` by making fragment a `String` [#642](https://github.com/rust-bitcoin/rust-miniscript/pull/642)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,30 @@
...
+- Remove unused generic on check_witness [#644](https://github.com/rust-bitcoin/rust-miniscript/pull/644)
```
Comment (storopoli) [inline CHANGELOG.md:18]:
```suggestion
- Remove unused generic on `check_witness` [#644](https://github.com/rust-bitcoin/rust-miniscript/pull/644)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,30 @@
...
+- Add conditional formatting for Terminal [#651](https://github.com/rust-bitcoin/rust-miniscript/pull/651)
```
Comment (storopoli) [inline CHANGELOG.md:19]:
```suggestion
- Add conditional formatting for `Terminal` [#651](https://github.com/rust-bitcoin/rust-miniscript/pull/651)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,30 @@
...
+
+## Other internal cleanups / improvements
+
+- Remove `internals` dependency [](https://github.com/rust-bitcoin/rust-miniscript/pull/631)
+- Introduce an example binary useful for profiling [#646](https://github.com/rust-bitcoin/rust-miniscript/pull/646)
+- Refactor out type_check [#649](https://github.com/rust-bitcoin/rust-miniscript/pull/649)
```
Comment (storopoli) [inline CHANGELOG.md:25]:
```suggestion
- Refactor out `type_check` [#649](https://github.com/rust-bitcoin/rust-miniscript/pull/649)
```

**Review by storopoli — COMMENTED** [review]
Since you've asked for eyes on the changelogs, I've added comments.
Most of these is for consistency throughout the new changelog.

**Comment by tcharding** [comment]
Thanks @storopoli, I'll re-spin with all your suggestions.

**Comment by tcharding** [comment]
Hey man to save you review time (which is precious) I would have accepted "you forgot to back tick all the types". But thanks for taking the time to point each out.

I also discovered a set of empty `[]` - so that's a win!

**Comment by tcharding** [comment]
All yours @apoelstra - review, ack, merge, tag, and publish please (if the changelog is acceptable).

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,30 @@
+# 12.0.0 - May 17, 2024
+
+- Update MSRV to Rust `v1.56.1` [#639](https://github.com/rust-bitcoin/rust-miniscript/pull/639)
+- Drop the `Property` trait entirely [#652](https://github.com/rust-bitcoin/rust-miniscript/pull/652)
+- Improve compiler logic when deciding between conjunctions and `multi`/`multi_a` [#657](https://github.com/rust-bitcoin/rust-miniscript/pull/657)
+- Several locktime improvements [#654](https://github.com/rust-bitcoin/rust-miniscript/pull/654)
+- Derive `Hash` for `pub` items [#659](https://github.com/rust-bitcoin/rust-miniscript/pull/659)
+- Introduce `Threshold` type [#660](https://github.com/rust-bitcoin/rust-miniscript/pull/660),
```
Comment (apoelstra) [inline CHANGELOG.md:8]: In 7ddaaf6f7f0d91e83eca659ccac340c9cda24883:

Trailing whitespace, and also a trailing comma.

Comment (tcharding) [reply CHANGELOG.md:8]: Fixed.

\me Puts coin in money jar for not reading diff before pushing

**Comment by tcharding** [comment]
Note, I did not rebase on `master` so this excludes the shellcheck CI job patch.

**Comment by tcharding** [comment]
Also, added #645 but did not add additional explanation - ok?

**Comment by tcharding** [comment]
```
# 12.0.0 - May 17, 2024

- Update MSRV to Rust `v1.56.1` [#639](https://github.com/rust-bitcoin/rust-miniscript/pull/639)
- Remove sketchy `LikelyFalse` error [#645](https://github.com/rust-bitcoin/rust-miniscript/pull/645)
```

**Comment by tcharding** [comment]
Updated date to 22nd

File: CHANGELOG.md (hunk 6 above)
Comment (storopoli) [reply CHANGELOG.md]: That jar is getting full... Reminds me a calculus teacher that had a jar of coins for everytime that a student forgot to add $c$ to an integration procedure 😂

Comment (tcharding) [reply CHANGELOG.md]: I had a swear jar once but it got full so quick I stopped. Now we do 5 pushups each time instead - works way better.

Comment (tcharding) [reply CHANGELOG.md]: Ha! Just did five push ups and said "read your diff" at the top of each one - lets see how it goes.

**Review by apoelstra — APPROVED** [review]
ACK 5ea33e389a848592d675e075ee7e63f756be69ed

**Comment by apoelstra** [comment]
Tagged and published.

**Pull request merged**